### DISCUSSION TITLE
Multiple modals loses state.

### DISCUSSION THREAD

**Discussion by BengtHagemeister** [mantinedev/mantine#3063] [Q&A]
Hi,

I have a modal with a form to create an entity. Inside the modal there is a possibility to open another modal to create child entities.

Unfortunately the modal manager is build in such a way that it completely removes the first modal from the DOM when the second modal is opened. Therefore losing all of its current state. Is there an easy way to prevent that?

**Comment by rtivital** [comment]
No, once modal is closed it is unmounted and the state is lost. You probably should use Modal component instead for this use case.

**Reply by ilhamgepe** [reply]
thanks its work. i use modal manager for submiting request post, but my state is lost. I don't know what the modal manager use for. but my problem solve when i use modal component instead of modal manager. thanks 🙏love mantine ❤️
